src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_IndustrialElectromagneticSeparator.java

```diff
@@ -322,25 +366,31 @@ public Collection<RecipeMap<?>> getAvailableRecipeMaps() {
     }
 
     @Override
-    public void saveNBTData(NBTTagCompound aNBT) {
-        aNBT.setBoolean("polarizerMode", polarizerMode);
-        super.saveNBTData(aNBT);
+    public boolean supportsMachineModeSwitch() {
+        return true;
     }
 
     @Override
-    public void loadNBTData(NBTTagCompound aNBT) {
-        polarizerMode = aNBT.getBoolean("polarizerMode");
-        super.loadNBTData(aNBT);
+    public void setMachineModeIcons() {
+        machineModeIcons.add(GT_UITextures.OVERLAY_BUTTON_MACHINEMODE_SEPARATOR);
+        machineModeIcons.add(GT_UITextures.OVERLAY_BUTTON_MACHINEMODE_POLARIZER);
+    }
+
+    @Override
+    public IMachineMode defaultMachineMode() {
+        return MachineMode.Separator;
     }
 
     @Override
     public void onScrewdriverRightClick(ForgeDirection side, EntityPlayer aPlayer, float aX, float aY, float aZ) {
-        polarizerMode = !polarizerMode;
-        if (polarizerMode) {
-            PlayerUtils.messagePlayer(aPlayer, "Now running in Polarizing Mode.");
-        } else {
-            PlayerUtils.messagePlayer(aPlayer, "Now running in Separating Mode.");
-        }
+        machineMode = machineMode.nextMachineMode();
+        PlayerUtils.messagePlayer(aPlayer, "Now running in " + machineMode.getModeName() + " Mode.");
+    }
+
+    @Override
+    public void loadNBTData(NBTTagCompound aNBT) {
+        if (machineMode == null) machineMode = MachineMode.Polarizer;
+        super.loadNBTData(aNBT);
```

Comment:
Don't forget about compatibility with existing saves. It would be annoying to reconfigure all machines *manually* after update.
Please, make sure that algorithm below is applied for all multi-mode machines.
```suggestion
        // Firstly, always try to read old tag to properly load existing saves.
        if (aNBT.hasKey("polarizerMode")) {
            machineMode = aNBT.getBoolean("polarizerMode") ? MachineMode.Polarizer :  MachineMode.Separator;
        }
        // Secondly, set mode to default one if previous step failed (and mode is still null).
        if (machineMode == null) {
            machineMode = MachineMode.Polarizer;
        }
        // Lastly, override mode from new tag. if such exists. That's already properly handled in base class, so just call:
        super.loadNBTData(aNBT);
```

Reply:
The machines I did not include this for were the MFE and TurboCan, which I had merged yesterday, so there is no real risk of anyone ever having built them - added it anyways, though. I guess there is no cost to doing this since loadNBT runs so rarely